manifests/params.pp

```diff
@@ -4,85 +4,101 @@
 #
 class augeasproviders::params {
 
-  $apache_directive_hash = $::augeasproviders_apache_directive_hash ? {
-    undef   => false,
-    default => $::augeasproviders_apache_directive_hash,
-  }
-
-  $apache_setenv_hash = $::augeasproviders_apache_setenv_hash ? {
-    undef   => false,
-    default => $::augeasproviders_apache_setenv_hash,
-  }
-
-  $host_hash = $::augeasproviders_host_hash ? {
-    undef   => false,
-    default => $::augeasproviders_host_hash,
-  }
-
-  $kernel_parameter_hash = $::augeasproviders_kernel_parameter_hash ? {
-    undef   => false,
-    default => $::augeasproviders_kernel_parameter_hash,
-  }
-
-  $mailalias_hash = $::augeasproviders_mailalias_hash ? {
-    undef   => false,
-    default => $::augeasproviders_mailalias_hash,
-  }
-
-  $mounttab_hash = $::augeasproviders_mounttab_hash ? {
-    undef   => false,
-    default => $::augeasproviders_mounttab_hash,
-  }
-
-  $nrpe_command_hash = $::augeasproviders_nrpe_command_hash ? {
-    undef   => false,
-    default => $::augeasproviders_nrpe_command_hash,
-  }
-
-  $pam_hash = $::augeasproviders_pam_hash ? {
-    undef   => false,
-    default => $::augeasproviders_pam_hash,
-  }
-
-  $pg_hba_hash = $::augeasproviders_pg_hba_hash ? {
-    undef   => false,
-    default => $::augeasproviders_pg_hba_hash,
-  }
-
-  $puppet_auth_hash = $::augeasproviders_puppet_auth_hash ? {
-    undef   => false,
-    default => $::augeasproviders_puppet_auth_hash,
-  }
-
-  $shellvar_hash = $::augeasproviders_shellvar_hash ? {
-    undef   => false,
-    default => $::augeasproviders_shellvar_hash,
-  }
-
-  $ssh_config_hash = $::augeasproviders_ssh_config_hash ? {
-    undef   => false,
-    default => $::augeasproviders_ssh_config_hash,
-  }
-
-  $sshd_config_hash = $::augeasproviders_sshd_config_hash ? {
-    undef   => false,
-    default => $::augeasproviders_sshd_config_hash,
-  }
-
-  $sshd_config_subsystem_hash = $::augeasproviders_sshd_config_subsystem_hash ? {
-    undef   => false,
-    default => $::augeasproviders_sshd_config_subsystem_hash,
-  }
-
-  $sysctl_hash = $::augeasproviders_sysctl_hash ? {
-    undef   => false,
-    default => $::augeasproviders_sysctl_hash,
-  }
-
-  $syslog_hash = $::augeasproviders_syslog_hash ? {
-    undef   => false,
-    default => $::augeasproviders_syslog_hash,
-  }
+  $apache_directive_hash = lookup({
+    'name'          => 'augeasproviders_apache_directive_hash',
+    'value_type'    => Hash,
+    'default_value' => {},
+  })
```

Comment:
Rather than these lookups, why don't we rely on the default lookups? So `augeasproviders::instances::apache_directive_hash` instead of `augeasproviders_apache_directive_hash`. It would also mean changing the instances to `Hash[String, Hash] $apache_directive_hash = {}`. Then you can drop most of this file.

Reply:
Exactly.